### PR TITLE
Add user defined IIT parameters to B-Tax

### PR DESCRIPTION
**Pull request by jdebacker**
This PR adds the ability for the user to define IIT parameters that affect the cost of capital and METR calculations in B-Tax.  These parameters are calculated from marginal tax rates found via the Tax Calculator.

### CODE REVIEW

File: btax/get_taxcalc_rates.py [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+    # long term capital gains
+    [mtr_fica_lcg, mtr_iit_lcg, mtr_combined_lcg] = calc1.mtr('p23250')
+    # pension distributions
+    # does PUF have e01500?  Do we want IRA distributions here?
+    # Weird - I see e01500 in PUF, but error when try to call it
+    [mtr_fica_pension, mtr_iit_pension, mtr_combined_pension] = calc1.mtr('e01700')
+    # mortgage interest and property tax deductions
+    # do we also want mtg ins premiums here?
+    # mtg interest
+    [mtr_fica_mtg, mtr_iit_mtg, mtr_combined_mtg] = calc1.mtr('e19200')
```
Comment (jdebacker) [inline btax/get_taxcalc_rates.py]: This line yields the following error:
`ValueError: mtr income_type_str="e19200" is not valid`

I see this variable in the PUF.  Am I not able to calculate the MTR w.r.t. this item with the Tax-Calculator?  Something else causing the error?

cc @MattHJensen @Amy-Xu

Comment (Amy-Xu) [inline btax/get_taxcalc_rates.py]: It seems from [this line of code](https://github.com/open-source-economics/Tax-Calculator/blob/master/taxcalc/calculate.py#L229) in Tax-Calculator that we only allow a certain set of variables to be passed into MTR function. This set currently does not include interest paid. We should be able to add this variable very easily. What do you think? @MattHJensen

Comment (jdebacker) [inline btax/get_taxcalc_rates.py]: Thanks for finding that @Amy-Xu.  I recall something like this, but couldn't find the lines of code to point out.  Does this limitation of the valid strings have to do with some TaxCalc logic or for another reason (e.g. to save computational time it does not compute MTRs for every line item on the 1040)?  

Relatedly, there are some variables such as tax exempt interest that is (obviously) zero-taxed under current law.  But one could imagine a reform where the tax-exempt status of this income source were to change.  How would one handle that?  In particular, would the Tax Calc output then put all interest income in  e00300?  Or would the formerly tax exempt still show up under e00400, but now face the non-zero tax?  I'm just trying to think about how I might compute a MTR on interest income as a whole (taking into account that some might be taxable and some not, with the taxable status changing between the baseline and reform).

cc @MattHJensen

Comment (MattHJensen) [inline btax/get_taxcalc_rates.py]: Just opened an issue in TC for this discussion. https://github.com/open-source-economics/Tax-Calculator/issues/880.

**Comment by jdebacker** [comment]
This PR still has one outstanding issue - the computation of MTRs related to owner occupied housing-related deductions.  These will inform the CC and METR calculations on investment in owner occupied housing.  This issue is being discussed in TC issue [#880](https://github.com/open-source-economics/Tax-Calculator/issues/880).

**Comment by jdebacker** [comment]
All IIT related parameters computed and baseline values look very close to those from [CBO (2014)](https://www.cbo.gov/sites/default/files/113th-congress-2013-2014/reports/49817-Taxing_Capital_Income_0.pdf).  Removing WIP label.  

@PeterDSteinberg @talumbau @brendancol  - any problem if I merge this to master with respect to your development of the UI?

**Comment by PeterDSteinberg** [comment]
@jdebacker Looks like the .travis.yml for this B-Tax repo needs to have the line `conda install -c ospc taxcalc` somewhere in pre-test section now that TaxCalc is required.

File: .travis.yml [hunk 2]
```diff
@@ -14,6 +14,7 @@ install:
   - conda create -n test-environment python=$TRAVIS_PYTHON_VERSION pandas numpy
   - source activate test-environment
   - conda install --file conda-requirements.txt # xlrd, possibly others
+  - conda install -c ospc taxcalc
```
Comment (jdebacker) [inline .travis.yml]: @PeterDSteinberg - Does this look ok?

FYI - the taxcalc version on conda is not yet able to handle what B-Tax does.  Those changes have only recently been incorporated into OSE/Tax-Calculator:master

Comment (PeterDSteinberg) [inline .travis.yml]: Yes that's good for longer-term.  In the near term we could do `git clone http:/github.com/jdebacker/TaxCalc && cd TaxCalc and git fetch --all && git checkout feature_branch && python setup.py install` , replacing feature_branch with the name of your branch of TaxCalc within your fork.

**Comment by jdebacker** [comment]
@PeterDSteinberg , let me know if I put the correct lines in the travis.yml file per your suggestion regarding the calling of the Tax Calculator.  If thats ok, I'll merge this PR.

**Comment by jdebacker** [comment]
@PeterDSteinberg , I've tried several things with the yml file and can't the command to change directory to the taxcalc directory to work.  I don't have a good understanding of what is going on here.  Can you advise?

**Comment by jdebacker** [comment]
@PeterDSteinberg , ok, got setup.py on taxcalc to run, but then getting error saying can't find setup.py for btax.  I think it's that I need to get out of the taxcalc directory and back to the btax directory.  But I'm unable to do that.  What is the directory structure?

File: .travis.yml [hunk 3]
```diff
@@ -19,7 +19,7 @@ install:
   - git clone https://github.com/jdebacker/Tax-Calculator.git && cd Tax-Calculator
   - git fetch --all && git checkout master && python setup.py install
```
Comment (PeterDSteinberg) [inline .travis.yml]: change the highlighted line from:
`git fetch --all && git checkout master && python setup.py install`
to
`git fetch --all && git checkout master && python setup.py install && cd ../B-Tax`

**Comment by jdebacker** [comment]
@PeterDSteinberg, Now I've got:

```
/home/travis/build.sh: line 45: cd: ../B-Tax: No such file or directory
The command "git fetch --all && git checkout master && python setup.py install && cd ../B-Tax" failed and exited with 1 during .
```

**Comment by PeterDSteinberg** [comment]
Let's try this:
```
  - git fetch --all && git checkout master && python setup.py install && ls . && ls .. && cd ..
```
That way we'll at least get some diagnostics if it fails.  If it's okay, then we can remove the `ls . && ls .. &&`

**Comment by jdebacker** [comment]
The current travis CI build fail has to do with missing puf.  I pushed btax/test_data/puf91taxdata.csv.gz, but still fail.  Does something need to be adjusted so taxcalc looks for test_data?

cc @PeterDSteinberg @talumbau

**Pull request merged**